### ISSUE TITLE
Add `hdd` and `ssd` required/forbid/optional (None) parameter

### ISSUE THREAD

**Issue by dandud100** [Nukesor/hetznerbot#16]
I would like to see the option to define the value for the search field in the config params.

**Comment by Nukesor** [comment]
Servers have no unique name and most properties are already covered by some parameter.
This service is designed to show you a range of products, I'm not exactly sure why you would need a search field?


Could you explain your use-case?

**Comment by daniel22-git** [comment]
I usually put the phrase "SSD" in it because it shows me all types of servers with an SSD and HDD.

If i use the other parameter in the search some servers are always excluded.

Like this search here, it shows every server with an 2TB HDD + any variations of ssd's.

https://www.hetzner.com/de/sb?ram_from=64&drives_size_from=2000&search=ssd

**Comment by Nukesor** [comment]
The thing is, that the search parameter is a generic search across all fields, that's implemented client side in hetzners bot.

Meaning, we would have to implement that search in this project as well, which is quite a lot of work and goes against the current approach of doing things, as you could no longer just query for matching offers by SQL.

My proposal would rather be, to add an `ssd_required` flag.

**Comment by daniel22-git** [comment]
> My proposal would rather be, to add an `ssd_required` flag.

Yeah if this is possible why not. I just want to get the same offers as from the shared link. 😁

**Comment by Nukesor** [comment]
Thinking about it, I would prefer a `hdd` and an `ssd` configs, which can have either of the states "None|require|forbid".
That would allow us to explicitly search for various combinations.


Feel free to start working on this. I would definitely accept this feature, if it would look like that :)

I won't build this myself, as I currently don't have the need for this.
